astropy/table/table.py

```diff
@@ -110,6 +110,12 @@ def __getitem__(self, item):
             raise IndexError('Illegal key or index value for {} object'
                              .format(self.__class__.__name__))
 
+    def __setitem__(self, item, value):
+        if item in self:
+            raise ValueError("Cannot replace column '{0}'.  Use Table.replace_column() instead."
+                             .format(item))
```

Comment:
Correct.  This is in `TableColumns`, so what is now disallowed is `t.columns['a'] = [1, 2, 3]`.  Previously this would not fail but the results would be unpredictable.